homeassistant/components/shelly/coordinator.py

```diff
@@ -97,7 +97,7 @@ def __init__(
             immediate=False,
             function=self._async_reload_entry,
         )
-        entry.async_on_unload(self._debounced_reload.async_cancel)
+        entry.async_on_unload(self._debounced_reload.async_shutdown)
```

Comment:
Contrary to `cancel`, `shutdown` also prevents new runs (for example in case of a failed reload which would trigger indefinitely).